src/System.Net.Http/src/System/Net/Http/StreamContent.cs

```diff
@@ -12,19 +11,30 @@ namespace System.Net.Http
 {
     public class StreamContent : HttpContent
     {
-        private const int defaultBufferSize = 4096;
+        private const int DefaultBufferSize = 4096;
 
         private Stream _content;
+        private CancellationToken _cancellationToken;
         private int _bufferSize;
         private bool _contentConsumed;
         private long _start;
 
         public StreamContent(Stream content)
-            : this(content, defaultBufferSize)
+            : this(content, DefaultBufferSize)
         {
         }
 
         public StreamContent(Stream content, int bufferSize)
+            : this(content, bufferSize, CancellationToken.None)
+        {
+        }
+
+        internal StreamContent(Stream content, CancellationToken cancellationToken)
```

Comment:
There is still time to adjust the 4.1 System.Net.Http contract.  I probably should have done it earlier.  But the SerializeToStreamAsync and CopyToAsync methods should have overloads to take a CancellationToken.  The SerializeToStreamAsync method is abstract and adding another abstract method to the HttpContent base class isn't allowed. But we could add a new virtual method overload of SerializeToStreamAsync to take a token. And the other *Async methods could take one as well.  That would be the best way to pass in a token.